### PR TITLE
More type hints

### PR DESCRIPTION
**Pull request by kevinhu**
Some prevailing issues:
- The stubs we're using for Pandas and Numpy from [data-science-types](https://github.com/predictive-analytics-lab/data-science-types) are incomplete or no up-to-date in certain cases, so we get a lot of errors from those.
- We pass around a bunch of complex nested dictionaries. For now, I've just used something like `Dict[str, Any]` for handling  these, but if we really want rigor, we should look into things like TypedDicts/dataclasses/etc. This would be a pretty huge commitment though.

### CODE REVIEW

File: ferry/config.py [hunk 1]
```diff
@@ -23,9 +26,15 @@
 # The latter three settings can also be set to a function. The function should return the
 # specified setting when called.
 CAS_USE_COOKIE = True
-CAS_CREDENTIAL_NETID = functools.lru_cache(lambda: input("Yale NetId: "))
-CAS_CREDENTIAL_PASSWORD = functools.lru_cache(getpass.getpass)
-CAS_COOKIE_CASTGC = functools.lru_cache(lambda: input("CASTGC Cookie: "))
+CAS_CREDENTIAL_NETID: "_lru_cache_wrapper[str]" = functools.lru_cache(
```
Comment (hsheth2) [inline ferry/config.py:29]: This annotation seems kinda ugly - can we not just make it a `Callable[[], str]`?

File: ferry/crawler/common_args.py [hunk 2]
```diff
@@ -33,7 +36,7 @@ def add_seasons_args(parser):
     )
 
 
-def parse_seasons_arg(arg_seasons, all_viable_seasons):
+def parse_seasons_arg(arg_seasons: List[Any], all_viable_seasons: List[Any]):
```
Comment (hsheth2) [inline ferry/crawler/common_args.py:39]: Should this be
```python
def parse_seasons_arg(arg_seasons: Optional[List[str]], all_viable_seasons: List[str]):
```

**Review by hsheth2 — CHANGES_REQUESTED** [review]
Adding all the types for those dicts isn't worth it.

I would love to get this to a stage where we can use this in CI. For now, we should run it in CI so we have the output but ignore all the errors (similar to what we do with eslint in the coursetable api)

File: ferry/config.py (hunk 1 above)
Comment (kevinhu) [reply ferry/config.py:29]: This type produces an error when it's overridden by config_private - for instance, with the cookie we have
```
ferry/config.py:43: error: Incompatible import of "CAS_COOKIE_CASTGC" 
(imported name has type "str", local name has type "Callable[[], str]")
```

Perhaps `Union[Callable[[], str], str]`?

Comment (hsheth2) [reply ferry/config.py:29]: Would rather that we change config_private - it's a bit more ugly on the config side but will make usage cleaner
```python
# config_private.py
CAS_COOKIE_CASTGC = lambda: "some secret value"
```

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**